### PR TITLE
Only run Typescript linting on Typescript files

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
Running TS linting on non Typescript files causes an error in Intellij (and probably elsewhere). This PR our eslint config to only run `eslint-config-typescript` against TS & TSX files

### CODE REVIEW

File: support-frontend/assets/helpers/abTests/abtest.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ function userInTest(
 	}
 
 	if (referrerControlled) {
-		return acquisitionDataTest && acquisitionDataTest.name === testId;
+		return !!acquisitionDataTest && acquisitionDataTest.name === testId;
```
Comment (marialani) [inline support-frontend/assets/helpers/abTests/abtest.ts:207]: Did you want this in here?

File: support-frontend/.eslintrc.js [hunk 2]
```diff
@@ -1,27 +1,34 @@
 module.exports = {
-  extends: [
-    '@guardian/eslint-config-typescript',
```
Comment (marialani) [inline support-frontend/.eslintrc.js:3]: Could we also remove this from our node modules

File: support-frontend/assets/helpers/abTests/abtest.ts (hunk 1 above)
Comment (rupertbates) [reply support-frontend/assets/helpers/abTests/abtest.ts:207]: Oops no, I think it will disappear when you merge your PR though!

**Comment by TomPretty** [comment]
This looks like it's added lots of `eslint` errors?

File: support-frontend/.eslintrc.js (hunk 2 above)
Comment (rupertbates) [reply support-frontend/.eslintrc.js:3]: As discussed it is still needed it's just moved lower in this file

**Review by marialani — APPROVED** [review]
Looks good!

**Pull request merged**